backend/coins/btc/electrum/electrum.go

```diff
@@ -25,44 +25,28 @@ import (
 
 	"github.com/digitalbitbox/bitbox-wallet-app/backend/coins/btc/blockchain"
 	"github.com/digitalbitbox/bitbox-wallet-app/backend/coins/btc/electrum/client"
+	"github.com/digitalbitbox/bitbox-wallet-app/backend/config"
 	"github.com/digitalbitbox/bitbox-wallet-app/util/errp"
 	"github.com/digitalbitbox/bitbox-wallet-app/util/jsonrpc"
-	"github.com/digitalbitbox/bitbox-wallet-app/util/rpc"
 	"github.com/digitalbitbox/bitbox-wallet-app/util/socksproxy"
 	"github.com/sirupsen/logrus"
 	"golang.org/x/net/proxy"
 )
 
-// Electrum holds information about the electrum backend
-type Electrum struct {
-	log        *logrus.Entry
-	serverInfo *rpc.ServerInfo
-	dialer     proxy.Dialer
-}
-
-// NewElectrum creates a new Electrum instance.
-func NewElectrum(log *logrus.Entry, serverInfo *rpc.ServerInfo, dialer proxy.Dialer) *Electrum {
-	return &Electrum{log, serverInfo, dialer}
-}
-
-// ServerInfo returns the server info for this backend.
-func (electrum *Electrum) ServerInfo() *rpc.ServerInfo {
-	return electrum.serverInfo
-}
-
-// EstablishConnection connects to a backend and returns an rpc client
+// establishConnection connects to a backend and returns an rpc client
 // or an error if the connection could not be established.
-func (electrum *Electrum) EstablishConnection() (io.ReadWriteCloser, error) {
+func establishConnection(
+	serverInfo *config.ServerInfo, dialer proxy.Dialer) (io.ReadWriteCloser, error) {
```

Comment:
80 or so characters isn't really Go's thing: https://github.com/golang/go/wiki/CodeReviewComments#line-length
I'd put it all on a single line.

Reply:
> There is no rigid line length limit in Go code, but avoid uncomfortably long lines.

In this project, it's not 80, but a "soft" rule of ~100. Imho not worth changing this instance as we have tons of line splits like this one. Should make it official with a style guideline, which will also include "no short variable names that have a bigger scope than a few lines" :laughing:

Reply:
It just makes it harder to read when you look at 1000s lines of a consistent style Go code and suddenly see this.

Reply:
(as in, 1000s lines in other random projects)

Reply:
Could've also been a `dial(conf *config.ServerInfo, dialer proxy.Dialer) (io.ReadWriteCloser, error)` which you'll most likely object due to the naming. :)

Reply:
Why would I object to `dial`? Sounds like a very good name ;)

Reply:
> It just makes it harder to read when you look at 1000s lines of a consistent style Go code and suddenly see this.

Eh, I am really past any formatting discussions. I wish `gofmt` would leave *zero* room for discussion, and every language had a `gofmt` equivalent. After `gofmt`, the things that remain hardly matter for readability compared to everything else, like architecture, structure, docs, tests, and so on.